examples/flutter_gallery/lib/demo/animation/home.dart

```diff
@@ -0,0 +1,533 @@
+  }) : _maxHeight = maxHeight, _scrollFactor = scrollFactor {
+    assert(maxHeight != null && maxHeight >= 0.0);
+    assert(scrollFactor != null && scrollFactor >= 1.0);
+  }
+
+  // The height of the status bar
+  double get maxHeight => _maxHeight;
+  double _maxHeight;
+  set maxHeight (double value) {
+    assert(maxHeight != null && maxHeight >= 0.0);
+    if (_maxHeight == value)
+      return;
+    _maxHeight = value;
+    markNeedsLayout();
+  }
+
+  // That rate at which this renderer's height shrinks when the scroll
+  // offset changes.
+  double get scrollFactor => _scrollFactor;
+  double _scrollFactor;
+  set scrollFactor (double value) {
+    assert(scrollFactor != null && scrollFactor >= 1.0);
+    if (_scrollFactor == value)
+      return;
+    _scrollFactor = value;
+    markNeedsLayout();
+  }
+
+  @override
+  void performLayout() {
+    final double height = (maxHeight - constraints.scrollOffset / scrollFactor).clamp(0.0, maxHeight);
+    geometry = new SliverGeometry(
+      paintExtent: math.min(height, constraints.remainingPaintExtent),
+      scrollExtent: maxHeight,
+      maxPaintExtent: maxHeight,
+    );
+  }
+}
+
+class _StatusBarPaddingSliver extends SingleChildRenderObjectWidget {
+  _StatusBarPaddingSliver({
+    Key key,
+    @required this.maxHeight,
+    this.scrollFactor: 5.0,
+  }) : super(key: key) {
+    assert(maxHeight != null && maxHeight >= 0.0);
+    assert(scrollFactor != null && scrollFactor >= 1.0);
+  }
+
+  final double maxHeight;
+  final double scrollFactor;
+
+  @override
+  _RenderStatusBarPaddingSliver createRenderObject(BuildContext context) {
+    return new _RenderStatusBarPaddingSliver(
+      maxHeight: maxHeight,
+      scrollFactor: scrollFactor,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderStatusBarPaddingSliver renderObject) {
+    renderObject.maxHeight = maxHeight;
+    renderObject.scrollFactor = scrollFactor;
+  }
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+    description.add('maxHeight: $maxHeight');
+    description.add('scrollFactor: $scrollFactor');
+  }
+}
+
+class _SliverAppBarDelegate extends SliverPersistentHeaderDelegate {
+  _SliverAppBarDelegate({
+    @required this.minHeight,
+    @required this.maxHeight,
+    @required this.child,
+  });
+
+  final double minHeight;
+  final double maxHeight;
+  final Widget child;
+
+  @override double get minExtent => minHeight;
+  @override double get maxExtent => math.max(maxHeight, minHeight);
+
+  @override
+  Widget build(BuildContext context, double shrinkOffset, bool overlapsContent) {
+    return new SizedBox.expand(
+      child: new Stack(
+        children: <Widget>[
+          child,
+          new Positioned(
+            top: 0.0,
+            left: 0.0,
+            child: new IconTheme(
+              data: new IconThemeData(color: Colors.white),
+              child: new BackButton(),
```

Comment:
I think the app bar plays some slight trick when sizing the back button.  I wonder if this is off by a few pixels horizontally.